service/resource/endpoint/create_test.go

```diff
@@ -1 +1,386 @@
+	}{
+		{
+			CurrentState: []Endpoint{
+				{
+					IP:               "1.1.1.1",
+					ServiceName:      "TestService",
+					ServiceNamespace: "TestNamespace",
+				},
+			},
+			DesiredState: []Endpoint{
+				{
+					IP:               "1.1.1.1",
+					ServiceName:      "TestService",
+					ServiceNamespace: "TestNamespace",
+				},
+			},
+			ExpectedCreateState: nil,
+		},
+		{
+			CurrentState: []Endpoint{
+				{
+					IP:               "1.1.1.1",
+					ServiceName:      "TestService",
+					ServiceNamespace: "TestNamespace",
+				},
+				{
+					IP:               "1.2.3.4",
+					ServiceName:      "TestService",
+					ServiceNamespace: "TestNamespace",
+				},
+			},
+			DesiredState: []Endpoint{
+				{
+					IP:               "1.1.1.1",
+					ServiceName:      "TestService",
+					ServiceNamespace: "TestNamespace",
+				},
+			},
+			ExpectedCreateState: nil,
+		},
+		{
+			CurrentState: []Endpoint{
+				{
+					IP:               "5.5.5.5",
+					ServiceName:      "TestService",
+					ServiceNamespace: "TestNamespace",
+				},
+				{
+					IP:               "1.2.3.4",
+					ServiceName:      "TestService",
+					ServiceNamespace: "TestNamespace",
+				},
+			},
+			DesiredState: []Endpoint{
+				{
+					IP:               "1.1.1.1",
+					ServiceName:      "TestService",
+					ServiceNamespace: "TestNamespace",
+				},
+			},
+			ExpectedCreateState: []Endpoint{
+				{
+					IP:               "1.1.1.1",
+					ServiceName:      "TestService",
+					ServiceNamespace: "TestNamespace",
+				},
+			},
+		},
+		{
+			CurrentState: nil,
+			DesiredState: []Endpoint{
+				{
+					IP:               "1.1.1.1",
+					ServiceName:      "TestService",
+					ServiceNamespace: "TestNamespace",
+				},
+			},
+			ExpectedCreateState: []Endpoint{
+				{
+					IP:               "1.1.1.1",
+					ServiceName:      "TestService",
+					ServiceNamespace: "TestNamespace",
+				},
+			},
+		},
+		{
+			CurrentState: []Endpoint{
+				{
+					IP:               "1.1.1.1",
+					ServiceName:      "TestService",
+					ServiceNamespace: "TestNamespace",
+				},
+			},
+			DesiredState:        nil,
+			ExpectedCreateState: nil,
+		},
+		{
+			CurrentState:        nil,
+			DesiredState:        nil,
+			ExpectedCreateState: nil,
+		},
+	}
+	var err error
+	var newResource *Resource
+	{
+		resourceConfig := DefaultConfig()
+		resourceConfig.K8sClient = fake.NewSimpleClientset()
+		resourceConfig.Logger = microloggertest.New()
+		newResource, err = New(resourceConfig)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
+	}
+	for i, tc := range testCases {
+		result, err := newResource.GetCreateState(context.TODO(), tc.Obj, tc.CurrentState, tc.DesiredState)
+		if err != nil {
+			t.Fatal("case", i+1, "expected", nil, "got", err)
+		}
+		if !reflect.DeepEqual(tc.ExpectedCreateState, result) {
+			t.Fatalf("case %d expected %#v got %#v", i+1, tc.ExpectedCreateState, result)
+		}
+	}
+}
+
+func Test_Resource_Endpoint_ProcessCreateState(t *testing.T) {
```

Comment:
These tests maybe don't cover enough yet, but checking the endpoint state is pretty tedious. Can this be done more efficiently? Otherwise I would not add more tests now and make some more progress.